### ISSUE TITLE
Animating the icon on transfer

### ISSUE THREAD

**Issue by es20490446e** [syncthing-gtk/syncthing-gtk#42]
Currently the tray icon only indicates when receiving files.

It would be useful if it indicated when sending them too, hence letting know that not all devices are synced.

As some devices, like phones, may not have any way of noticing that.

Rather than spinning, it would be more useful if it changed color for both. For example blue when sending or receiving, and green when everything is done for all detected devices.

**Comment by mikoducky** [comment]
I'd love a better tray icon in general. I remember on Manjaro how the tray was a nice blue that was grey when it was doing nothing. Now it's an ugly PS1 era looking black or white with no transparency.

Currently running Fedora 38 on xorg

**Comment by es20490446e** [comment]
At least it should look not low resolution.

**Comment by Rudd-O** [comment]
GNOME has adopted the black and white aesthetic for indicator icons.  I agree the icon should indicate activity.

The mustard indicates progress.

**Comment by mikoducky** [comment]
I thought GNOME killed tray icons. So who cares what they think?

**Comment by es20490446e** [comment]
Probably black and white in this instance would be problematic, as color more clearly indicates the state of sync.

If it is black and white, for example, maybe you won't notice a special state.

**Comment by Rudd-O** [comment]
Indicators still appear on top of the GNOME panel.  This tech does not support traditional tray icons.

**Comment by Rudd-O** [comment]
> Probably black and white in this instance would be problematic, as color more clearly indicates the state of sync.

Our opinion doesn't matter, the HIG is clear that the icons must be monochromatic.

**Comment by mikoducky** [comment]
well then maybe just make the tray icon an svg or something right in the main folder so each individual user can fix GNOME's poor styling decisions. Or back to my original point: just make the icon better instead of looking like the game save icon for a ps1 memory card block

**Comment by es20490446e** [comment]
> Our opinion doesn't matter, the HIG is clear that the icons must be monochromatic.

I think it refers only to the actual GNOME desktop icons.

Because anything else is show in a hidden sub-menu, like [this](https://i.imgur.com/V7r3RlH.png). Isn't it?

Although this desktop design seems rather broken. Because like that I don't see the use of a tray icon anymore, as it indicates nothing.

**Comment by Rudd-O** [comment]
> I think it refers only to the actual GNOME desktop icons.

No, these are not monochromatic.  UI icons are monochromatic per HIG: https://developer.gnome.org/hig/guidelines/ui-icons.html .  They do it so the platform can recolor them without impact to the quality of the displayed result.

**Comment by QmwJlHuSg9pa** [comment]
> No, these are not monochromatic. UI icons are monochromatic per HIG: https://developer.gnome.org/hig/guidelines/ui-icons.html . They do it so the platform can recolor them without impact to the quality of the displayed result.

That page doesn't explain things very well. I do see: "All or part of the icon can be programmatically recolored." Though, web searching "gnome hig programmatically recolor icon" isn't very fruitful.

Then, further in the page, it links to [Symbolic Preview](https://flathub.org/apps/org.gnome.design.SymbolicPreview).

Luckily, the first preview image for this app shows a single (partially) coloured icon, which seems to represent a missed call:
![](https://dl.flathub.org/repo/screenshots/org.gnome.design.SymbolicPreview-stable/752x423/org.gnome.design.SymbolicPreview-4bd8c403b128a0108f6e700c16ac4e43.png)

If the intention is that the hypothetical icon for this sync app is to be monochrome while inactive, with highlights or solid shading to represent state (e.g. blue/green/red to represent transfer/success/fail), then that sounds alright to me.
